### PR TITLE
feat(workflow) added a workflow to check all 3rd party licences from all MFEs

### PR DESCRIPTION
**Pull request by ArtieReus**
## Description
This PR:
- Added workflow to check all 3rd party licenses from all MFEs living in `ui` folder
- Adds an apache badge to all greenhouse MFEs since it is Juno standard and triggers the workflow describe above.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [x] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [x] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .github/workflows/3rd-party-licenses.yaml [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+#  3. Create a pull_request.json file in case of a pull request with the following content:
+#     `{"pull_request": {"number": <PR number>, "head": {"ref": "<PR branch name>", "sha": "PR commit sha"}, "base": {"ref": "main"}}, "repository": {"name": "greenhouse", "owner": {"login": "cloudoperators"}}}`
+#  4. Run the following command:
+# `act pull_request --container-architecture linux/amd64 -P default=catthehacker/ubuntu:act-latest  -j check-3rd-party-licenses -e act_pull_request.json`
+
+name: 3rd Party License Checker
+
+on:
+  pull_request:
+    types: [opened, synchronize, reopened]
```
Comment (kengou) [inline .github/workflows/3rd-party-licenses.yaml:15]:
```suggestion
    types: [opened, synchronize, reopened]
    paths:
      - 'ui/**'
```

Comment (kengou) [reply .github/workflows/3rd-party-licenses.yaml:15]: I would add a filter here to run it only when `ui` elements changed

File: .github/workflows/3rd-party-licenses.yaml [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+          echo changed files: ${{ steps.filters.outputs.packages_files}}
+          echo package changes: ${{ steps.packages-filters.outputs.changes}}          
+          echo "===================="
+
+  check-3rd-party-licenses:
+    needs: detect-changes
+    runs-on: [default]
+    strategy:
+      matrix:
+        change: ${{fromJson(needs.detect-changes.outputs.changes)}}
```
Comment (kengou) [inline .github/workflows/3rd-party-licenses.yaml:61]: I propose to move the node version here , to be able to test it in the future for multiple node version if we ever need it 

```suggestion
        change: ${{fromJson(needs.detect-changes.outputs.changes)}}
        node: [ 20.x ]
```

File: .github/workflows/3rd-party-licenses.yaml [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup node
+        uses: actions/setup-node@v4
+        with:
+          node-version: "20.x"
```
Comment (kengou) [inline .github/workflows/3rd-party-licenses.yaml:69]:
```suggestion
          node-version: ${{ matrix.node }}
```

**Review by hgw77 — APPROVED** [review]

**Review by kengou — CHANGES_REQUESTED** [review]
I added some nitpicks

File: .github/workflows/3rd-party-licenses.yaml (hunk 1 above)
Comment (ArtieReus) [reply .github/workflows/3rd-party-licenses.yaml:15]: yes, this is a good idea

File: .github/workflows/3rd-party-licenses.yaml (hunk 2 above)
Comment (ArtieReus) [reply .github/workflows/3rd-party-licenses.yaml:61]: This is probably not needed, but it doesnt hurt... :)

**Pull request merged**